h2o-core/src/main/java/water/ExternalFrameUtils.java

```diff
@@ -46,57 +52,87 @@ public static ByteChannel getConnection(String ipPort) throws IOException{
-            }else if(clazz == Short.class){
-                expectedTypes[i] = EXPECTED_SHORT;
-            }else if(clazz == Character.class){
-                expectedTypes[i] = EXPECTED_CHAR;
-            }else if(clazz == Integer.class){
-                expectedTypes[i] = EXPECTED_INT;
-            }else if(clazz == Long.class){
-                expectedTypes[i] = EXPECTED_LONG;
-            }else if(clazz == Float.class){
-                expectedTypes[i] = EXPECTED_FLOAT;
-            }else if(clazz == Double.class){
-                expectedTypes[i] = EXPECTED_DOUBLE;
-            }else if(clazz == String.class){
-                expectedTypes[i] = EXPECTED_STRING;
-            }else if(clazz == Timestamp.class){
-                expectedTypes[i] = EXPECTED_TIMESTAMP;
-            }else{
-                throw new IllegalArgumentException("Unknown java class " + clazz);
+
+        byte[] vecTypes = new byte[size];
+        int vectorCount = 0;
+        int currentVecIdx = 0;
+        for (byte expectedType: expectedTypes) {
+            switch (expectedType) {
+                case EXPECTED_BOOL:
+                    vecTypes[currentVecIdx] = Vec.T_NUM;
+                    currentVecIdx++;
+                    break;
+                case EXPECTED_BYTE:
+                    vecTypes[currentVecIdx] = Vec.T_NUM;
+                    currentVecIdx++;
+                    break;
+                case EXPECTED_CHAR:
+                    vecTypes[currentVecIdx] = Vec.T_NUM;
+                    currentVecIdx++;
+                    break;
+                case EXPECTED_SHORT:
+                    vecTypes[currentVecIdx] = Vec.T_NUM;
+                    currentVecIdx++;
+                    break;
+                case EXPECTED_INT:
+                    vecTypes[currentVecIdx] = Vec.T_NUM;
+                    currentVecIdx++;
+                    break;
+                case EXPECTED_LONG:
+                    vecTypes[currentVecIdx] = Vec.T_NUM;
+                    currentVecIdx++;
+                    break;
+                case EXPECTED_FLOAT:
+                    vecTypes[currentVecIdx] = Vec.T_NUM;
+                    currentVecIdx++;
+                    break;
+                case EXPECTED_DOUBLE:
+                    vecTypes[currentVecIdx] = Vec.T_NUM;
+                    currentVecIdx++;
+                    break;
+                case EXPECTED_STRING:
+                    vecTypes[currentVecIdx] = Vec.T_STR;
+                    currentVecIdx++;
+                    break;
+                case EXPECTED_TIMESTAMP:
+                    vecTypes[currentVecIdx] = Vec.T_TIME;
+                    currentVecIdx++;
+                    break;
+                case EXPECTED_VECTOR:
+                    for (int j = 0; j < vecElemSizes[vectorCount]; j++) {
+                        vecTypes[currentVecIdx] = Vec.T_NUM;
+                        currentVecIdx++;
+                        break;
```

Comment:
This is problem! We really need lot of tests for this stuff to avoid this kind of mistakes,